### PR TITLE
GEODE-3321: Adding ErrorCode values to protobuf protocol

### PR DESCRIPTION
**Pull request by WireBaron**
@pivotal-amurmann @kohlmu-pivotal @hiteshk25 @galen-pivotal @bschuchardt 

Signed-off-by: Bruce Schuchardt <bschuchardt@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-protobuf/src/main/java/org/apache/geode/protocol/protobuf/ProtocolErrorCode.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing), software distributed under the
+ * License is distributed on an "AS IS" BASIS), WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND),
+ * either express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.geode.protocol.protobuf;
+
+public enum ProtocolErrorCode {
+  GENERIC_FAILURE(1000),
```
Comment (galen-pivotal) [inline geode-protobuf/src/main/java/org/apache/geode/protocol/protobuf/ProtocolErrorCode.java]: Why are we starting at 1000?

File: geode-protobuf/src/test/java/org/apache/geode/protocol/protobuf/operations/RemoveRequestOperationHandlerJUnitTest.java [hunk 2]
```diff
@@ -95,8 +97,8 @@ public void processReturnsUnsucessfulResponseForInvalidRegion()
         operationHandler.process(serializationServiceStub, removeRequest, cacheStub);
 
     assertTrue(result instanceof Failure);
-    org.junit.Assert.assertThat(result.getErrorMessage().getMessage(),
-        CoreMatchers.containsString("Region"));
+    assertEquals(ProtocolErrorCode.REGION_NOT_FOUND.codeValue,
```
Comment (galen-pivotal) [inline geode-protobuf/src/test/java/org/apache/geode/protocol/protobuf/operations/RemoveRequestOperationHandlerJUnitTest.java]: YES. I like these kinds of tests much better.

**Comment by galen-pivotal** [comment]
It looks like this is failing in Travis due to `:rat`, but you've got a license in your only new file. I'm a bit mystified. Closing the pull request and reopening it can be a good way to trigger a new Travis build.

**Comment by bschuchardt** [comment]
The license header was probably damaged during some search&replace operations to transform the google-doc version of the list into an enum

File: geode-protobuf/src/main/java/org/apache/geode/protocol/protobuf/ProtocolErrorCode.java (hunk 1 above)
Comment (pivotal-amurmann) [inline geode-protobuf/src/main/java/org/apache/geode/protocol/protobuf/ProtocolErrorCode.java]: This originally started at "1" and was followed by "100". After I protested it was changed to "001". After more discussion we ended up with the "1000" and "2000" ranges to accommodate for sub-ranges.

**Comment by WireBaron** [comment]
Copied the license fresh from another file.

File: geode-protobuf/src/test/java/org/apache/geode/protocol/protobuf/operations/RemoveRequestOperationHandlerJUnitTest.java (hunk 2 above)
Comment (kohlmu-pivotal) [inline geode-protobuf/src/test/java/org/apache/geode/protocol/protobuf/operations/RemoveRequestOperationHandlerJUnitTest.java]: +1

File: geode-protobuf/src/main/java/org/apache/geode/protocol/protobuf/ProtocolErrorCode.java (hunk 1 above)
Comment (kohlmu-pivotal) [inline geode-protobuf/src/main/java/org/apache/geode/protocol/protobuf/ProtocolErrorCode.java]: I believe any error code structure decides how many characters are to be used for an error code. 
The fact that we represent them as integers is merely a "simplification" choice over using Alpha numerics.
I don't believe starting the error code at 1000 is a problem

Comment (galen-pivotal) [inline geode-protobuf/src/main/java/org/apache/geode/protocol/protobuf/ProtocolErrorCode.java]: sure, just curious.

**Comment by galen-pivotal** [comment]
Looks like there are some conflicts that need to be resolved.

**Comment by WireBaron** [comment]
Rebased to fix conflicts and added an error code to the new GetAvailableServers handler.

**Pull request closed without merging**